File: tqcommon.py
from configparser import ConfigParser
import os
import socket
import tomli
import yaml


def get_env() -> str:
    return "dev" if socket.gethostname().startswith("d") else "prod"


def get_rstar_dir() -> str:
    return os.path.join(os.sep, "content", get_env(), "rstar")


def get_sysconfig():
    env = get_env()
    etcdir = f"/content/{env}/rstar/etc"
    conf_file = f"{etcdir}/task-queue.sysconfig"

    config = {}
    if os.path.isfile(conf_file):
        with open(conf_file) as fh:
            for line in fh:
                line = line.partition("#")[0].strip()
                if line:
                    k, v = line.split("=")
                    config[k.lower()] = v
    return config


def get_myconfig_file():
    return f"/content/{get_env()}/rstar/etc/my-taskqueue.cnf"


def get_myconfig():
    conf_file = get_myconfig_file()
    config = ConfigParser()
    config.read(conf_file)
    return dict(config["client"])


def get_hpc_config():
    conf_file = os.path.join(get_rstar_dir(), "etc", "hpc-taskqueue.toml")
    with open(conf_file, "rb") as f:
        config = tomli.load(f)
    return config["main"]


def get_host_aliases():
    alias_file = f"/content/{get_env()}/rstar/etc/host-aliases.yaml"
    aliases = {}
    if os.path.exists(alias_file):
        with open(alias_file) as fh:
            aliases = yaml.safe_load(fh)["aliases"]
    return aliases

File: lib/tqcommon.rb
# frozen_string_literal: true

require 'resolv'
require 'socket'
require 'yaml'

# Provides common functionality for the task queue
module TQCommon
  INSTALL_DIR = '/usr/local/dlib/task-queue'

  module_function

  def env
    /^d/ =~ Socket.gethostname ? 'dev' : 'prod'
  end

  def rstar_dir
    "/content/#{env}/rstar"
  end

  def tmpdir
    "#{rstar_dir}/tmp"
  end

  def sysconfig
    conf_file = "/content/#{env}/rstar/etc/task-queue.sysconfig"
    config = {}
    File.foreach(conf_file) do |line|
      line.strip!
      next if line.empty? || line.start_with?('#')

      key, value = line.split('=', 2).map(&:strip)
      config[key.downcase] = value
    end
    config
  end

  def host_aliases
    alias_file = "/content/#{env}/rstar/etc/host-aliases.yaml"
    aliases = {}
    if File.exist?(alias_file)
      aliases = YAML.load_file(alias_file)['aliases']
      aliases = aliases.transform_keys { |name| name[/^[^.]+/] }
    end
    aliases
  end

  def services
    # services_file = "/content/#{env}/rstar/etc/services.yaml"
    services_file = File.expand_path('../services.yaml', File.dirname(__FILE__))
    YAML.load_file(services_file)['services']
  end

  def smtp_host
    config = sysconfig
    config.fetch('smtphost', 'localhost')
  end

  def hostname(host)
    if /\A(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})\Z/ =~ host
      begin
        host = Resolv.getname(host)[/^[^.]+/]&.downcase || host
      rescue Resolv::ResolvError
        nil
      end
    end
    host
  end
end
